app/models/extraction_definition.rb

```diff
@@ -64,4 +67,17 @@ def to_h
   def shared?
     harvest_definitions.count > 1
   end
+
+  def clone(pipeline, name)
+    cloned_extraction_definition = ExtractionDefinition.new(dup.attributes.merge(name:, pipeline:))
+
+    requests.each do |request|
+      cloned_request = request.dup
```

Comment:
Might be not understanding, but I wonder if delegating what it means to clone a request would be better put on the request model (if that is a model)

Reply:
Yeah I was wondering about this too. It's probably fine here but I don't feel too strongly about it.